libsolidity/interface/CompilerStack.cpp

```diff
@@ -858,7 +866,7 @@ string CompilerStack::applyRemapping(string const& _path, string const& _context
 
 void CompilerStack::resolveImports()
 {
-	solAssert(m_stackState == ParsingSuccessful, "");
+	if (!m_parserErrorRecovery) solAssert(m_stackState == ParsingSuccessful, "");
```

Comment:
This really gets too complicated. It might be better to rename the states to `ParsingPerformed` and `AnalysisPerformed` and then have an additional flag that tells whether there are errors (or maybe the errorList already does that).

Reply:
The existings tests in [CompilerStack.h](https://github.com/ethereum/solidity/blob/develop/libsolidity/interface/CompilerStack.h#L83-L90) are: 

```C++
	enum State {
		Empty,
		SourcesSet,
		ParsingSuccessful,
		AnalysisSuccessful,
		CompilationSuccessful
	};
```

Is  this more in line with what would be better here? : 

```C++
	enum State {
		Empty,
		SourcesSet,
                ParsingPerformed,   // Parsing done, but possibly has errors
		ParsingSuccessful,  // Parsing done and has no errors
                AnalysisPerformed,  // Analysis done in the face of parsing errors
		AnalysisSuccessful, // Analysis done and no parsing or analysis errors
		CompilationSuccessful
	};
```
If so, I'll split this out. If not suggest a concrete change to the Enum.

Reply:
No, I meant it as I said above - "rename".

Reply:
Ok. So I gather then: 

```
enum State {
		Empty,
		SourcesSet,
                ParsingPerformed,   // Parsing done, but possibly has errors
                AnalysisPerformed,  // Analysis done in the face of parsing errors
		CompilationSuccessful
	};
```

and booleans for `ParsingSuccessful` and `AnalysisSuccessful`. Is that correct?

Reply:
Is there a need to discriminate where the error happened?

Reply:
I don't think so, but I don't want to commit to any decision that doesn't need to be committed to. Once we get a fully working pipeline going it will be easier to discuss and describe what and why.

I don't see how the question relates to the enum set or the booleans. Please be more expansive.

Reply:
We do need to commit to a change. If we introduce a (or multiple) boolean, we need to commit to the meaning of that boolean. My suggestion would be to more or less keep the enum as it is (just changing the wording a you wrote above) and have a single boolean that prevents the state transition to compilation.

Reply:
Ok - thanks for the clarification. Will look at when I get back to this.

Reply:
Done in 10bf0cac32490efdfb98334991b5e1cbe63d68a2 now.